src/test/java/org/example/CRUDAPITests.java

```diff
@@ -0,0 +1,173 @@
+        activity.setDueDate(dueDate);
+        activity.setCompleted(true);
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .body(activity)
+                .when()
+                .post(baseUrl);
+
+        response.then().assertThat().statusCode(200)
+                .and().body("title", equalTo(title))
+                .and().body("dueDate", equalTo(dueDate));
+    }
+
+    @Test
+    public void testReadAll() {
+        int expectedNumberOfRecords = 30;
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .when()
+                .get(baseUrl);
+
+        response.then().assertThat().statusCode(200)
+                .and().body("size()", equalTo(expectedNumberOfRecords));
+    }
+
+    @Test
+    public void testReadById() {
+        int id = 1;
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .pathParam("id", id)
+                .when()
+                .get(baseUrl + "/{id}");
+        response.then().assertThat().statusCode(200);
+    }
+
+    @Test
+    public void testUpdate() {
+        int id = 1;
+        String title = "new title";
+        String dueDate = "2023-11-29T17:52:03.204Z";
+        boolean completed = false;
+
+        Activity activity = new Activity();
+        activity.setId(id);
+        activity.setTitle(title);
+        activity.setDueDate(dueDate);
+        activity.setCompleted(completed);
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .body(activity)
+                .pathParam("id", id)
+                .when()
+                .put(baseUrl + "/{id}");
+
+        response.then().assertThat().statusCode(200)
+                .and().body("title", equalTo(title))
+                .and().body("dueDate", equalTo(dueDate))
+                .and().body("completed", equalTo(completed));
+    }
+
+    @Test
+    public void testDelete() {
+        int id = 1;
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .pathParam("id", id)
+                .when()
+                .delete(baseUrl + "/{id}");
+        response.then().assertThat().statusCode(200);
+        response = given()
+                .contentType(ContentType.JSON)
+                .pathParam("id", id)
+                .when()
+                .get(baseUrl + "/{id}");
+        response.then().assertThat().statusCode(404);
+    }
+
+    @Test
+    public void testCreateNewActivityApi_Negative() {
+        String title = "string";
+        String dueDate = "2023-11-29T17:52:03.204Z";
+
+        Activity activity = new Activity();
+        activity.setId(0);
+        activity.setTitle(title);
+        activity.setDueDate(dueDate);
+        activity.setCompleted(true);
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .body(activity)
```

Comment:
I guess the body is superfluous here with full object fields, u can try pass just 'new Activity()', as you  just call the invalid url path.

Reply:
Test Fixed